index.js

```diff
@@ -107,8 +107,9 @@ function getSequelizeTypeFromJoi(dialect, type, rules) {
 
         return Sequelize.TEXT;
     case 'array':
-    case 'object':
         return Sequelize.TEXT;
+    case 'object':
+        return Sequelize.TEXT('medium');
```

Comment:
If the database doesn't support any variant of `TEXT`, it just skip with warn log.
https://github.com/sequelize/sequelize/blob/73f1b5d1d3cd010286e4db5fdb7e1baee22cd1f6/lib/dialects/postgres/data-types.js#L103

Reply:
Also confirmed that sqlite dialect do the same thing as postgres
https://github.com/sequelize/sequelize/blob/73f1b5d1d3cd010286e4db5fdb7e1baee22cd1f6/lib/dialects/sqlite/data-types.js#L80